### PR TITLE
Display empty string instead of <nil> when IP opt is nil.

### LINKED ISSUE #13878: Nil IP addresses options are displayed as "<nil>" in the help message
I just noticed that `docker --help` displays `<nil>` for options (flags) that have a type of `IPVar' and an empty default:

```shell
% docker --help | grep nil
  --default-gateway=<nil>              Container default gateway IPv4 address
  --default-gateway-v6=<nil>           Container default gateway IPv6 address
```

I think it would be better not to display anything.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #13878.

### CODE REVIEW

**Comment by runcom** [comment]
this is weird and seems a regression cause I'm not experiencing the problem on 1.7.0-rc2 :/
@eolamey could you please add a integration test?

**Comment by eolamey** [comment]
It is indeed a regression from 1.7.0-rc2, and comes from https://github.com/docker/docker/pull/13757, for which I am guilty :) The flag type was changed from `flag.StringVar` to `opts.IPVar`. On the bright side, `--default-gateway` doesn't work at all in 1.7.0-rc2 :)

It is not a regression in `opts.IPVar` per se, as there was no flag with a nil IP address before.

Should I try to add an integration test, or a unit test in `opts/opts_test.go`?

**Comment by runcom** [comment]
@eolamey unit test make sense thanks

**Comment by eolamey** [comment]
@runcom, does this look ok to you?

**Comment by runcom** [comment]
Yep, assuming janky is happy this LGTM

File: opts/opts_test.go [hunk 1]
```diff
@@ -179,6 +180,18 @@ func TestValidateExtraHosts(t *testing.T) {
 	}
 }
 
+func TestIpOptString(t *testing.T) {
+	addresses := []string{"", "0.0.0.0"}
```
Comment (duglin) [inline opts/opts_test.go]: Should we add a non-empty case too?

Comment (icecrime) [inline opts/opts_test.go]: I'm not sure I understand your comment, there's both `""` and `"0.0.0.0"` (surely non valid, but yet non-empty).

**Comment by icecrime** [comment]
LGTM

File: opts/opts_test.go (hunk 1 above)
Comment (duglin) [inline opts/opts_test.go]: doi - sorry, I misread the test. For some reason I thought that was just one test where 0.0.0.0 ended up mapping to "".
One day I'll learn to read  :-)

Comment (icecrime) [inline opts/opts_test.go]: Reading is overrated!

**Comment by duglin** [comment]
LGTM

**Comment by jfrazelle** [comment]
cherry-picked

**Pull request merged**